### PR TITLE
feat: add ScatterElements operator support

### LINKED ISSUE #125: Support ScatterElements operator in ONNX import/export
### Feature description

Implement support for the ScatterElements ONNX operator in both import and export routines within burn-onnx. This operator allows multidimensional scatter updates along an axis and is frequently used in models exported from recent PyTorch, ONNX, and other ML frameworks.

### Feature motivation

- ScatterElements is a core ONNX operator recently used in computer vision and language models for advanced indexing and batched elementwise updates.
- Recent PyTorch, ONNX, and TensorFlow ONNX exports favor ScatterElements for masking, sequence editing, and shape management tasks.
- Burn has a robust backend scatter implementation (see: https://github.com/tracel-ai/burn/search?q=scatter&type=code ), so this operator should be mapped to Burn's API for compatibility.

### (Optional) Suggest a Solution

- Parse and map ONNX ScatterElements (https://onnx.ai/onnx/operators/onnx__ScatterElements.html) to the Burn scatter API for all compatible tensor types, supporting arbitrary axis.
- Validate support for gradient, shape, and dtype coverage as in other arithmetic ops.
- Add round-trip and conversion tests for import/export from ONNX models using ScatterElements.

References:
- [Burn scatter implementation](https://github.com/tracel-ai/burn/search?q=scatter&type=code)
- [ONNX ScatterElements operator spec](https://onnx.ai/onnx/operators/onnx__ScatterElements.html)

### PR DESCRIPTION
**Pull request by antimora**
## Summary

- Implement ScatterElements ONNX operator (opset 11+) with all 5 reduction modes: none, add, mul, max, min
- `add` reduction uses Burn's native `scatter()` with `IndexingUpdateOp::Add`
- `none` reduction (non-Bool) uses an optimized gather-diff-scatter approach: `data.scatter(dim, indices, updates - data.gather(dim, indices), Add)`
- `mul`/`max`/`min` and Bool `none` use element-by-element loop codegen (pending upstream Burn support for additional `IndexingUpdateOp` variants: https://github.com/tracel-ai/burn/issues/4477)

## Test plan

- [x] 11 unit tests for onnx-ir processor (config extraction, axis normalization, reduction parsing, type inference)
- [x] 8 insta snapshot tests for burn-onnx codegen (all reduction modes, int tensors, bool tensors)
- [x] 6 integration tests against ONNX models (none/axis0, axis1, add, mul, max, min)
- [x] Full test suite passes (457 tests, 0 failures)
- [x] No clippy warnings

Closes #125

### CODE REVIEW

File: crates/burn-onnx/src/burn/node/scatter_elements.rs [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+                            result,
+                        );
+                    },
+                    ScatterElementsReduction::Add => unreachable!(),
+                };
+
+                quote! {
+                    let #output = {
+                        let data_dims = #data.dims();
+                        let updates_dims = #updates.dims();
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/burn-onnx/src/burn/node/scatter_elements.rs:92]: This is codegen (the code shown here is emitted into the generated `.rs` file). Adding an `eprintln\!` to every forward pass would spam stderr on every inference call. The limitation is already documented in the PR description with a link to the upstream Burn issue (https://github.com/tracel-ai/burn/issues/4477) for adding native `IndexingUpdateOp` variants. Once Burn supports `Assign`/`Mul`/`Max`/`Min`, this fallback path goes away entirely.

File: crates/burn-onnx/src/burn/node/scatter_elements.rs [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+                        let dim: usize = #dim;
+
+                        let mut data_strides = alloc::vec![1usize; r];
+                        for i in (0..r.saturating_sub(1)).rev() {
+                            data_strides[i] = data_strides[i + 1] * data_dims[i + 1];
+                        }
+                        let mut idx_strides = alloc::vec![1usize; r];
+                        for i in (0..r.saturating_sub(1)).rev() {
+                            idx_strides[i] = idx_strides[i + 1] * updates_dims[i + 1];
+                        }
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/burn-onnx/src/burn/node/scatter_elements.rs:105]: The ONNX spec requires data, indices, and updates to all have the same rank. This is validated by the ONNX model checker before it reaches our code, and onnx-ir type inference enforces 3 inputs with matching tensor types. A malformed model with mismatched ranks would fail earlier in the pipeline. Adding a runtime rank check in the generated code would be defensive against something that cannot happen with valid ONNX models.

File: crates/burn-onnx/src/burn/node/scatter_elements.rs [hunk 3]
```diff
@@ -0,0 +1,511 @@
...
+
+                        for flat_idx in 0..total_updates {
+                            let mut remaining = flat_idx;
+                            let mut target_offset = 0usize;
+                            for d in 0..r {
+                                let coord = remaining / idx_strides[d];
+                                remaining %= idx_strides[d];
+                                if d == dim {
+                                    let mut idx = indices_values[flat_idx];
+                                    if idx < 0 { idx += data_dims[d] as i64; }
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/burn-onnx/src/burn/node/scatter_elements.rs]: Good catch. Added a bounds check with a clear error message in 07c24741. Without it, an out-of-bounds index would produce a cryptic panic from `narrow`/`slice_assign` instead of a meaningful error.
